ws/ws.nim

```diff
@@ -260,7 +304,7 @@ proc handleRequest*(
 proc new*(
   _: typedesc[WSServer],
   protos: openArray[string] = [""],
-  factories: openArray[ExtFactory] = [],
+  factories: openArray[(string, ExtFactory)] = [],
```

Comment:
Was thinking about this. Maybe just pass a `Table` instead (not sure)?

Reply:
I have strong reason to keep the `openArray` interface.

1. not sacrifice ergonomics. I imagine authors of extensions would export their extensions using:
```Nim
const
    deflateExt* = ("permessage-deflate", deflateFactory)

const
    snappyExt* = ("permessage-snappy", snappyFactory)

etc..

let ws = WSServer.new(factories = [deflateExt, snappyExt])  # still ergonomic
```

2. flexibility to swap internal implementation
I assume in real life production code, the number of extensions is not very much. probably not much than 5.
If that is the case, it is more economically to use plain `seq` instead of `Table`. Table is overkill if number of exts is small.
keeping the `openArray` interface will allow us to change the implementation without breaking the interface.

Reply:
> I have strong reason to keep the openArray interface.

OK, I like the reasoning and I don't really have a strong objection against this.

> If that is the case, it is more economically to use plain seq instead of Table. Table is overkill if number of exts is small.

Yeah agree with that, I believe I had it as a `seq` initially?

Reply:
yes, and I will change it back to seq.